lib/apiservers/service/restapi/handlers/vch_create_test.go

```diff
@@ -0,0 +1,296 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	cli "gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/govmomi/list"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/cmd/vic-machine/common"
+	"github.com/vmware/vic/cmd/vic-machine/create"
+	"github.com/vmware/vic/lib/apiservers/service/models"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+type mockFinder struct {
+	path string
+}
+
+func (mf mockFinder) Element(ctx context.Context, t types.ManagedObjectReference) (*list.Element, error) {
+	return &list.Element{
+		Path: t.Value,
+	}, nil
+}
+
+func TestFromManagedObject(t *testing.T) {
+	op := trace.NewOperation(context.Background(), "TestFromManagedObject")
+	var m *models.ManagedObject
+
+	expected := ""
+	actual, err := fromManagedObject(op, nil, "t", m)
+	assert.NoError(t, err, "Expected nil error, got %#v", err)
+	assert.Equal(t, expected, actual, "Expected %s, got %s", expected, actual)
+
+	m = &models.ManagedObject{
+		Name: "testManagedObject",
+	}
+
+	mf := mockFinder{}
+
+	expected = m.Name
+	actual, err = fromManagedObject(op, mf, "t", m)
+	assert.NoError(t, err, "Expected nil error, got %#v", err)
+	assert.Equal(t, expected, actual, "Expected %s, got %s", expected, actual)
+
+	m.ID = "testID"
```

Comment:
This doesn't clear the `Name`. Should there be a separate test case for having an ID and no name?

Reply:
That's actually a good point - currently, the func will return `Name` if `ID` is `nil`. So we could update this to test expect an empty string in that case, if that's the intended behavior.

Reply:
Good enough for now :+1: